src/components/OrdersWidget/OrderRow.tsx

```diff
@@ -152,44 +91,156 @@ function calculatePrice(_numerator?: string | null, _denominator?: string | null
   return price.toFixed(2)
 }
 
-const OrderRow: React.FC<Props> = props => {
-  const { buyToken, sellToken, order, pending = false } = props
+interface OrderDetailsProps extends Pick<Props, 'order' | 'pending'> {
+  buyToken: TokenDetails
+  sellToken: TokenDetails
+}
 
+const OrderDetails: React.FC<OrderDetailsProps> = ({ buyToken, sellToken, order, pending }) => {
   const price = useMemo(
     () =>
       calculatePrice(
         formatAmountFull(order.priceNumerator, buyToken.decimals, false),
         formatAmountFull(order.priceDenominator, sellToken.decimals, false),
       ),
-    [buyToken.decimals, order.priceDenominator, order.priceNumerator, sellToken.decimals],
+    [buyToken, order.priceDenominator, order.priceNumerator, sellToken],
   )
+
+  return (
+    <div className="container order-details">
+      <div>Sell</div>
+      <div>
+        <Highlight color={pending ? 'grey' : ''}>1</Highlight>
+      </div>
+      <div>
+        <strong>{displayTokenSymbolOrLink(sellToken)}</strong>
+      </div>
+
+      <div>
+        for <strong>at least</strong>
+      </div>
+      <div>
+        <Highlight color={pending ? 'grey' : 'red'}>{price}</Highlight>
+      </div>
+      <div>
+        <strong>{displayTokenSymbolOrLink(buyToken)}</strong>
+      </div>
+    </div>
+  )
+}
+
+interface UnfilledAmountProps extends Pick<Props, 'order' | 'pending'> {
+  sellToken: TokenDetails
+}
+
+const UnfilledAmount: React.FC<UnfilledAmountProps> = ({ sellToken, order, pending }) => {
   const unfilledAmount = useMemo(() => formatAmount(order.remainingAmount, sellToken.decimals) || '0', [
     order.remainingAmount,
     sellToken.decimals,
   ])
+  const unlimited = useMemo(() => order.priceDenominator.gt(MIN_UNLIMITED_SELL_ORDER), [order.priceDenominator])
```

Comment:
This kind of check is very small and produces primitive data, so I wouldn't memoize it

Reply:
Yep, agree with this comment. Memoizing makes it a bit more difficult to read, for fast operations that don't hit the network I wouldn't worry about performance issues until they are notizable